Classes/Controller/PageViewController.php

```diff
@@ -88,9 +138,277 @@ public function mainAction(): void
+                ['tx_dlf[multipleSource][' . $nextMultipleSourceKey . ']' => $formAddDocument->getLocation()],
+                ['tx_dlf[multiview]' => 1]
+            );
+            $uriBuilder = $this->uriBuilder;
+            $uri = $uriBuilder
+                ->setArguments($params)
+                ->setArgumentPrefix('tx_dlf')
+                ->uriFor('main');
+
+            $this->redirectToUri($uri);
+        }
+
+    }
+
+    /**
+     * Get all measures from musical struct
+     * @param int $page
+     * @param ?AbstractDocument $specificDoc
+     * @return array
+     */
+    protected function getMeasures(int $page, $specificDoc = null, $docNumber = null): array
+    {
+        if ($specificDoc) {
+            $doc = $specificDoc;
+        } else {
+            $doc = $this->document->getCurrentDocument();
+        }
+        $currentPhysId = $doc->physicalStructure[$page];
+        $measureCoordsFromCurrentSite = [];
+        $measureCounterToMeasureId = [];
+        $measureLinks = [];
+        $defaultFileId = $doc->physicalStructureInfo[$currentPhysId]['files']['DEFAULT'];
+        if (isset($defaultFileId)) {
+            $musicalStruct = $doc->musicalStructureInfo;
+
+            $i = 0;
+            foreach ($musicalStruct as $measureData) {
+                if ($defaultFileId == $measureData['files']['DEFAULT']['fileid']) {
+                    $measureCoordsFromCurrentSite[$measureData['files']['SCORE']['begin']] = $measureData['files']['DEFAULT']['coords'];
+                    $measureCounterToMeasureId[$i] = $measureData['files']['SCORE']['begin'];
+
+                    if ($specificDoc) {
+                        // build link for each measure
+                        $params = [
+                            'tx_dlf' => $this->requestData,
+                            'tx_dlf[docMeasure][' . $docNumber . ']' => $i
+                        ];
+                    } else {
+                        // build link for each measure
+                        $params = [
+                            'tx_dlf' => $this->requestData,
+                            'tx_dlf[measure]' => $i
+                        ];
+                    }
+                    $uriBuilder = $this->uriBuilder;
+                    $uri = $uriBuilder
+                        ->setArguments($params)
+                        ->setArgumentPrefix('tx_dlf')
+                        ->uriFor('main');
+                    $measureLinks[$measureData['files']['SCORE']['begin']] = $uri;
+
+                }
+                $i++;
+            }
+        }
+        return [
+            'measureCoordsCurrentSite' => $measureCoordsFromCurrentSite,
+            'measureCounterToMeasureId' => $measureCounterToMeasureId,
+            'measureLinks' => $measureLinks
+        ];
+    }
+
+    /**
+     * Get score URL and MIME type
+     *
+     * @access protected
+     *
+     * @param int $page: Page number
+     * @param ?AbstractDocument $specificDoc
```

Comment:
```suggestion
     * @param ?MetsDocument $specificDoc
```
This only works for `MetsDocument`, not for any `AbstractDocument`, because method `getPageBeginning()` is only available in `MetsDocument`.

This should fix the following `Call to an undefined method` error.